byron/ledger/executable-spec/test/Ledger/UTxO/Properties.hs

```diff
@@ -29,6 +34,44 @@ moneyIsConstant = withTests 200 . property $ do
   (st0, st) <- firstAndLastState <$> forAll (trace @UTXOW 500)
   reserves st0 + balance (utxo st0) === reserves st + balance (utxo st)
 
+-- | Check that a UTxO has no common inputs with a transaction
+noCommonInputs :: UTxO -> Tx -> Bool
+noCommonInputs utxo0 tx = dom (txouts tx) ∩ dom utxo0 == empty
+
+-- | Check that there is no double spending
+noDoubleSpending :: Property
+noDoubleSpending = property $ do
+  t <- forAll (trace @UTXO 500)
+  let
+    utxo0 = fst . firstAndLastState $ t :: State UTXO
+    txs = traceSignals OldestFirst t :: [Signal UTXO]
+  when (all (noCommonInputs $ utxo utxo0) txs) $ do
+    (all (noCommonInputsTxs txs) (zip txs [0 .. ])) === True
+   where
+    noCommonInputsTxs :: [Tx] -> (Tx, Int) -> Bool
+    noCommonInputsTxs txs (tx, i) =
+      let ins = txins' tx
+      in all (\txj -> txins' txj ∩ ins == empty) (take i txs)
+
+    txins' :: Tx -> Set TxIn
+    txins' = fromList . txins
+
+-- | Check that UTxO is outputs minus inputs
+utxoDiff :: Property
+utxoDiff = property $ do
+  t <- forAll (trace @UTXO 500)
+  let
+    (utxo0, utxoSt) = (\(f, s) -> (utxo f, utxo s)) . firstAndLastState $ t :: (UTxO, UTxO)
+    txs = traceSignals OldestFirst t :: [Signal UTXO]
+  when (all (noCommonInputs utxo0) txs) $ do
```

Comment:
There is a redundant `do` and a couple of redundant parentheses. You might want to run `hlint` on the file.

Reply:
Thanks! Fixed this in the upcoming commit.